src/components/grouped-character/grounded-character.test.tsx

```diff
@@ -0,0 +1,604 @@
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(5);
+  expect(input.selectionEnd).toBe(5);
+
+  await user.keyboard("1");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(7);
+  expect(input.selectionEnd).toBe(7);
+});
+
+test("when no character is present, cursor position goes to 0", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(0);
+  expect(input.selectionEnd).toBe(0);
+});
+
+test("pressing backspace when the cursor is after the middle character in the final group, should move the cursor one position backwards", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("1234567");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 3]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
```

Comment:
**suggestion**: it seems we don’t need to exhaust the fake timers after a userEvent action, since the action is already wrapped with `act` to keep React in sync with the UI. What do you think about removing those cases?

```suggestion
```